kpet/run.py

```diff
@@ -160,6 +160,9 @@ def __get_hosts(database, target):
                     pool_suites.remove(pool_suite)
             # Add host to list, if it has suites to run
             if suites:
+                suites = list(sorted(suites, key=lambda suite:
+                                     [case for case in suite.cases if
+                                      case.waived] != []))
```

Comment:
I wonder if it would be a bit better to sort by the number of waived tests. As in:
```
list(sorted(suites, key=lambda suite:
            len([case for case in suite.cases if case.waived])))
```
That way suites that have only one waived would go before suites that have many. Also, personally I find it a bit unnatural to compare lists in Python... but that's entirely subjective.

Also, isn't it a bit weird to align the `[case` with `suites`, and not `key=`?

Reply:
I also think this way of sorting them seems cleaner but it does change the result and I'm not sure there's a reason to keep them in the same order (after splitting them to waived/ not waived)

The alignment thing would seem to make sense to align with key= but that would make the line too long and PEP doesn't like it either (E127)

Reply:
Hmm, to sort by the number of waived tests might be a good idea. Let's see...

Yup, Ondrej is right about the alignment.